### ISSUE TITLE
Improve error message for config typos

### ISSUE THREAD

**Issue by samuelbroscheit** [uma-pi1/kge#79]
For a config like 

```
job.type: train
dataset.name: toy
model: complex

complex:
  relation_embedders: # <--- typo relation_embedder*s* 
    regularize_args:
      p: 1
```

the error message is very technical 

`KeyError: 'complex.relation_embedders.regularize_args cannot be set because creation of complex.relation_embedders is not permitted'`

which could be improved to say 

`KeyError: 'Key "complex.relation_embedders" does not exist. Parent key "complex" does not allow creation of new keys. If the creation of "complex.relation_embedders" was intended, then "complex" should have the "+++" attribute.'`

**Comment by rgemulla** [comment]
Or simply:

```
Key "complex.relation_embedders.regularize" cannot be set because key "complex.relation_embedders" does not exist and no new keys are allowed to be created under "complex".
```

It's important that the message says which key was supposed to be set first.

**Comment by rgemulla** [comment]
Addressed in c3e3ba05398abccec3f95674e014d34524ec4486
